File: common/src/main/scala/com/zto/bigdata/spark/common/rest/Rest.scala
package com.zto.bigdata.spark.common.rest

import spark.{Request, Response}

/**
  * 用于封装rest的相关信息
  *
  * @param method
  * rest的提交方式：GET/POST/PUT/DELETE等
  * @param path
  * rest服务地址
  * @param funw
  * rest的处理逻辑
  * @author ChengLong 2019-3-16 09:58:06
  */
case class Rest(method: String, path: String, fun: (Request, Response) => AnyRef)

File: fire-common/src/main/java/com/zto/fire/common/anno/Rest.java
package com.zto.fire.common.anno;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 用于标识启用restful接口
 *
 * @author ChengLong 2019-4-16 11:07:13
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.FIELD})
public @interface Rest {

    /**
     * restful路径名
     *
     * @return
     */
    String value() default "";

    /**
     * 接口访问的方式: GET/POST
     *
     * @return
     */
    String method() default "GET";
}
